src/Services/TemplatedEmailService/TemplatedEmailService.cs

```diff
@@ -42,15 +42,22 @@ public Task ProcessTemplatedEmail(List<IAction> actions)
             var formData = _distributedCache.GetString(sessionGuid);
             var formAnswers = JsonConvert.DeserializeObject<FormAnswers>(formData);
 
-
             foreach (var action in actions)
             {
                 var templatedEmailProvider = _templatedEmailProviders.Get(action.Properties.EmailTemplateProvider);
+                var convertedAnswers = formAnswers.AllAnswers.ToDictionary(x => x.QuestionId, x => x.Response);
+                var personalisation = new Dictionary<string, dynamic>();
+
+                if (action.Properties.IncludeCaseReference)
+                    personalisation.Add("reference", formAnswers.CaseReference);
 
+                if (!convertedAnswers.Count.Equals(0) && action.Properties.Personalisation is not null)          
```

Comment:
&& will evaulate the first part of this if first, convertedAnswers could be null and so accessing .Count first will throw an exception. Shoul de checking if its null first.